### PR TITLE
Finish all timers

### PR DESCRIPTION
**Pull request by Jameskmonger**
https://neworbit.tpondemand.com/entity/41796-user-can-finish-all-timers

### CODE REVIEW

File: src/register-commands.ts [hunk 1]
```diff
@@ -37,6 +39,11 @@ const offsetFlagConfig = {
     description: "specify a positive number of days in the past"
 };
 
+const allFlagConfig = {
+    flags: "--all",
+    description: "finish all of a days timers"
```
Comment (thyde1) [inline src/register-commands.ts]: Grammar pedantry: "finish all of a day's timers"

File: src/commands/prompts/finish.ts [hunk 2]
```diff
@@ -70,23 +51,52 @@ const askTimeRemaining = async (tpEntity: any, timeEntry: HarvestTimeEntry) => {
...
-        tpEntity,
-        timeEntry,
-        timeRemaining
-    };
+    for (const timeEntry of timeEntries) {
+        const tpEntity = await getTargetprocessEntity(targetprocessApi, timeEntry.notes.entity.id);
+
+        const timeRemaining = await askTimeRemaining(tpEntity, timeEntry);
+
+        finishDetails.push({
```
Comment (thyde1) [inline src/commands/prompts/finish.ts]: I assume the targetProcess Api deals nicely with timeRemaining being `NaN`?

File: src/register-commands.ts (hunk 1 above)
Comment (Jameskmonger) [reply src/register-commands.ts]: Fixed

File: src/commands/prompts/finish.ts (hunk 2 above)
Comment (Jameskmonger) [reply src/commands/prompts/finish.ts]: `askTimeRemaining` checks that `isNaN` is false before parsing

**Pull request merged**